text_generation/causal_lm/cpp/continuous_batching/library/src/sequence_group.hpp

```diff
@@ -445,12 +446,30 @@ class SequenceGroup {
             }
         // For greedy or multinomial sampling we decide whever to stream partial results depending on the user parameter
         } else if (m_sampling_params.is_greedy_sampling() || m_sampling_params.is_multinomial()) {
-            // TO DO: Now we always stream for greedy search for the sake of benchmarking
-            if (true /* m_sampling_params.stream */) {
+            // TO DO: Now we always stream for greedy search for the sake of benchmarking 
+            if (num_total_seqs() == 1 /* m_sampling_params.stream */) {
+                // TODO: support streamimg for n seqs
                 for (auto& sequence : m_sequences) {
-                        outputs.emplace(sequence->get_grouped_id(), sequence->get_last_generation_output(m_sampling_params));
+                    // todo: check seq.is_finished() to generate without several </s>
+                    // or is it ok to use padding?
+                    const auto last_gen_token = sequence->get_last_generation_output();
+                    outputs.emplace(sequence->get_grouped_id(), last_gen_token);
```

Comment:
Why do we break it into two lines? Is it for some future processing?

Reply:
Because it looks prettier as for me. Anyway, not a blocker

Reply:
Okay, I'm fine with it. Was just curious if there's a further plan for `last_gen_token` 😄